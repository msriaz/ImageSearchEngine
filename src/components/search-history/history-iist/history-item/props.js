import PropTypes from 'prop-types';

export const propTypes = {
  title: PropTypes.string.isRequired,
  onPressHistoryItem: PropTypes.func.isRequired,
};

export const defaultProps = {};
