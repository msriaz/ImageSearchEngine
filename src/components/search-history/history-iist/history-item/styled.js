import styled from 'styled-components/native';

export const StyledWrapper = styled.TouchableOpacity`
  padding: ${({theme}) => theme.spacing.small}px;
  background-color: ${({theme}) => theme.color.lightestGray};
  border-radius: ${({theme}) => theme.radius.half}px;
  margin: ${({theme}) => theme.spacing.xsmall}px;
  flex-direction: row;
  justify-content: space-between;
`;

export const StyledItem = styled.Text`
  font-size: 14px;
  font-weight: 200;
  margin: ${({theme}) => theme.spacing.xsmall}px;
`;
