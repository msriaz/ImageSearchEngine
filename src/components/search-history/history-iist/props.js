import PropTypes from 'prop-types';

export const propTypes = {
  searchQuery: PropTypes.string.isRequired,
  onPressHistoryItem: PropTypes.func.isRequired,
};

export const defaultProps = {};
