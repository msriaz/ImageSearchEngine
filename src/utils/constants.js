import {color as colors} from '../theme';
import {generateColorPalleteConstants} from './index';

export const APP_COLORS = generateColorPalleteConstants(colors);

export const STORAGE_KEYS = {
  HISTORY: 'history',
};
