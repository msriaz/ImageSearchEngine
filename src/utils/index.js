import {useEffect, useRef} from 'react';
import snakeCase from 'lodash/snakeCase';
/**
 *
 * @param {*} value
 * @returns previous state
 */
export const usePrevious = value => {
  const ref = useRef();
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
};

/**
 *
 * @param {*} imgData
 * @returns image url to render
 */
export const getImageUrl = imgData => {
  const {farm, server, id, secret} = imgData;
  return `https://farm${farm}.static.flickr.com/${server}/${id}_${secret}.jpg`;
};

/**
 * generate color consts from color pallette
 * @param {*} colors
 */
export const generateColorPalleteConstants = colors => {
  const colorsList = {};
  Object.keys(colors).forEach(colorKey => {
    colorsList[snakeCase(colorKey).toUpperCase()] = colors[colorKey];
  });
  return colorsList;
};

/**
 *
 * @param {*} funcName
 * @param {*} error
 */
export const reportError = (funcName, error) => {
  console.log(`Error in ${funcName}`, error);
};
