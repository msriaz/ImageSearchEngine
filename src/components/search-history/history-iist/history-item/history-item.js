import React from 'react';
import {propTypes, defaultProps} from './props';

import {StyledWrapper, StyledItem} from './styled';
/**
 * Image Item Component
 * @param {title} defines the title of history item
 * @param {onPressHistoryItem} defines the action to be called on pressing history item
 */
export const HistoryItem = ({title, onPressHistoryItem}) => {
  const onPressHistoryItemHandler = () => onPressHistoryItem(title);
  return (
    <StyledWrapper onPress={onPressHistoryItemHandler}>
      <StyledItem>{title}</StyledItem>
    </StyledWrapper>
  );
};

HistoryItem.propTypes = propTypes;
HistoryItem.defaultProps = defaultProps;
