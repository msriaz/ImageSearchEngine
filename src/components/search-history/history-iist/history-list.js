import React, {forwardRef} from 'react';
import {FlatList} from 'react-native';

import {HistoryItem} from './history-item';

/**
 * Image Item Component
 * @param {data} defines the data of images to render
 * @param {loadMore} defines the function to be called for loading more data
 * @param {rest} defines additional props to pass to FlatList
 */
export const HistoryList = ({data, onPressHistoryItem, rest}) => {
  const renderItem = ({item}) => (
    <HistoryItem
      title={item}
      key={item}
      onPressHistoryItem={onPressHistoryItem}
    />
  );

  return (
    <FlatList data={data} renderItem={renderItem} {...rest} />
  );
};
