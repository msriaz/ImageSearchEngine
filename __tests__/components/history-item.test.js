import React from 'react';
import {fireEvent} from '@testing-library/react-native';

import {HistoryItem} from '../../src/components/search-history/history-iist/history-item';
import {render} from '../utils/theme-wrapper';

const onPressMock = jest.fn();
const mockItem = 'test item';

describe('History Item Component', () => {
  it('should match the History Item  snapshot', () => {
    const compTree = render(
      <HistoryItem title={mockItem} onPressHistoryItem={onPressMock} />,
    );
    expect(compTree.toJSON()).toMatchSnapshot();
  });

  it('History Item should be displayed', () => {
    const {getByText} = render(
      <HistoryItem title={mockItem} onPressHistoryItem={onPressMock} />,
    );
    fireEvent.press(getByText(mockItem));
    expect(onPressMock).toHaveBeenCalled();
  });
});
