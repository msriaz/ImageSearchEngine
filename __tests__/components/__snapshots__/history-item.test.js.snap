// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Item Component should match the History Item  snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F4F4F4",
      "borderRadius": 30,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "200",
        },
      ]
    }
  >
    test item
  </Text>
</View>
`;
